shared_ptr usage in eclipse writer is broken`with g++-4.4

```
[  0%] Building CXX object CMakeFiles/opmcore.dir/opm/core/io/eclipse/EclipseWriter.cpp.o
/home/mblatt/src/dune/opm/opm-core/opm/core/io/eclipse/EclipseWriter.cpp:1303: warning: unused parameter ‘wellState’
/home/mblatt/src/dune/opm/opm-core/opm/core/io/eclipse/EclipseWriter.cpp: In constructor ‘Opm::EclipseWriter::EclipseWriter(const Opm::parameter::ParameterGroup&, std::shared_ptr<const Opm::EclipseGridParser>, std::shared_ptr<const UnstructuredGrid>)’:
/home/mblatt/src/dune/opm/opm-core/opm/core/io/eclipse/EclipseWriter.cpp:1543: error: no matching function for call to ‘std::shared_ptr<const Opm::Deck>::shared_ptr(int)’
/usr/include/c++/4.4/bits/shared_ptr.h:1263: note: candidates are: std::shared_ptr<_Tp>::shared_ptr(std::shared_ptr<_Tp>&&) [with _Tp = const Opm::Deck]
/usr/include/c++/4.4/bits/shared_ptr.h:1238: note:                 std::shared_ptr<_Tp>::shared_ptr() [with _Tp = const Opm::Deck]
/usr/include/c++/4.4/bits/shared_ptr.h:1236: note:                 std::shared_ptr<const Opm::Deck>::shared_ptr(const std::shared_ptr<const Opm::Deck>&)
```

The problem seems to be the statements

```
parser_(0)
```

I will provide a patch shortly

--- a/opm/core/io/eclipse/EclipseWriter.cpp
+++ b/opm/core/io/eclipse/EclipseWriter.cpp
@@ -1538,7 +1538,7 @@ EclipseWriter::EclipseWriter (
         std::shared_ptr <const EclipseGridParser> parser,
         std::shared_ptr <const UnstructuredGrid> grid)
     : parser_ (parser)
-    , newParserDeck_(0)
+    , newParserDeck_()
     , grid_ (grid)
     , uses_ (phaseUsageFromDeck (*parser))
 {
@@ -1592,7 +1592,7 @@ EclipseWriter::EclipseWriter (
         const ParameterGroup& params,
         Opm::DeckConstPtr newParserDeck,
         std::shared_ptr <const UnstructuredGrid> grid)
-    : parser_ (0)
+    : parser_ ()
     , newParserDeck_(newParserDeck)
     , grid_ (grid)
     , uses_ (phaseUsageFromDeck (newParserDeck))

